services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_specs.py

```diff
@@ -138,10 +139,10 @@ def _update_resource_limits_and_reservations(
 
         if docker_compose_major_version >= 3:
             # compos spec version 3 and beyond
-            deploy = spec.get("deploy", {})
-            resources = deploy.get("resources", {})
-            limits = resources.get("limits", {})
-            reservations = resources.get("reservations", {})
+            deploy: dict[str, Any] = spec.get("deploy", {})
+            resources_v3: dict[str, Any] = deploy.get("resources", {})
```

Comment:
_v3???

Reply:
this is to highlight that from dcoker-compose > 3, `limits` and `reservations` etc are grouped under `resources`. `resources` is a very generic name and i thought this might highlight that. In addition, this entry is also mapped into `deploy.resources` of the `specs`, so it has "double use"

Reply:
but then probably you should rename more than half of the code in there.. not sure that make sense. I think the deploy part is also very much changed from v2 to v3. do we even support v2? not sure. maybe we should better define this and then if we only support v3, which I think is probably the case, remove it.

Reply:
yes, if you see the entire code, there is a part that supports v2

Reply:
yes but my point is that the whole deploy part is probably v3 only https://nickjanetakis.com/blog/docker-tip-51-which-docker-compose-api-version-should-you-use#:~:text=Some%20of%20the%20major%20differences,ll%20need%20to%20use%20v3%2B.
do we really support v2?

Reply:
yes V2 is also supported, in the else block below.
I would not rename the variable here. The `if else` statement encapsulates this. At most create 2 functions to wrap this code for v2 and v3

Reply:
let it be.